File: app/src/main/java/rssdemo/andras/hu/rssdemo/ui/Navigator.java
package rssdemo.andras.hu.rssdemo.ui;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;

import rssdemo.andras.hu.rssdemo.ui.subscriptions.SubscriptionsActivity;

public final class Navigator {

    private Context context;

    public Navigator(Context context) {
        this.context = context;
    }

    public void goToSubscriptionScreen() {
        Intent intent = new Intent(context, SubscriptionsActivity.class);
        context.startActivity(intent);
    }

    public void goToUrl(String url) {
        Intent i = new Intent(Intent.ACTION_VIEW);
        i.setData(Uri.parse(url));
        context.startActivity(i);
    }
}

File: app/src/main/java/rssdemo/andras/hu/rssdemo/ui/Navigator.kt
package rssdemo.andras.hu.rssdemo.ui

import android.content.Context
import android.content.Intent
import android.net.Uri

import rssdemo.andras.hu.rssdemo.ui.subscriptions.SubscriptionsActivity

class Navigator(private val context: Context) {

    fun goToSubscriptionScreen() {
        val intent = Intent(context, SubscriptionsActivity::class.java)
        context.startActivity(intent)
    }

    fun goToUrl(url: String) {
        val i = Intent(Intent.ACTION_VIEW)
        i.data = Uri.parse(url)
        context.startActivity(i)
    }
}
